photutils/psf/photometry.py

```diff
@@ -407,3 +355,361 @@ def _model_params2table(self, fit_model, star_group):
+
+            # n_start = 2 because it starts in the second iteration
+            # since the first iteration is above
+            output_table = self._do_photometry(n_start=2)
+            output_table = vstack([table, output_table])
+        else:
+            output_table = self._do_photometry()
+        return output_table
+
+    def _do_photometry(self, n_start=1):
+        """
+        Helper function which performs the iterations of the photometry process.
+
+        Parameters
+        ----------
+        n_start : int
+            Integer representing the start index of the iteration.
+            It is 1 if positions are None, and 2 otherwise.
+
+        Returns
+        -------
+        output_table : `~astropy.table.Table` or None
+            Table with the photometry results, i.e., centroids and
+            fluxes estimations and the initial estimates used to start
+            the fitting process.
+            None is returned if no sources are found in ``image``.
+        """
+
+        output_table = Table([[], [], [], [], [], [], [], [], []],
+                names=('x_0', 'y_0', 'flux_0', 'id', 'group_id',
+                       'iter_detected', 'x_fit', 'y_fit',
+                       'flux_fit'),
+                dtype=('f8', 'f8', 'f8', 'i4', 'i4', 'i4', 'f8', 'f8',
+                       'f8'))
+        sources = self.finder(self._residual_image)
+
+        n = n_start
+        while(len(sources) > 0 and
+              (self.niters is None or n <= self.niters)):
+            apertures = CircularAperture((sources['xcentroid'],
+                                          sources['ycentroid']),
+                                          r=self.aperture_radius)
+            sources['aperture_flux'] = aperture_photometry(self._residual_image,
+                    apertures)['aperture_sum']
+            init_guess_tab = Table(names=['x_0', 'y_0', 'flux_0'],
+                               data=[sources['xcentroid'],
+                               sources['ycentroid'],
+                               sources['aperture_flux']])
+            table = super(IterativelySubtractedPSFPhotometry,
+                    self).do_photometry(self._residual_image, init_guess_tab)
+            table['iter_detected'] = n*np.ones(table['x_fit'].shape, dtype=np.int)
+            output_table = vstack([output_table, table])
+            sources = self.finder(self._residual_image)
+            n += 1
+
+        return output_table
+
+
+class DAOPhotPSFPhotometry(IterativelySubtractedPSFPhotometry):
+    """
+    This class implements  an iterative algorithm based on the
+    DAOPHOT algorithm presented by Stetson (1987) to perform point
+    spread function photometry in crowded fields. This consists of
+    applying a loop of find sources, make groups, fit groups, subtract
+    groups, and then repeat until no more stars are detected or a given
+    number of iterations is reached.
+
```

Comment:
I would add something here to make it just a bit clearer how it connects to `IterativelySubtractedPSFPhotometry`.  Like maybe a sentence at the end of the first paragraph along the lines of "this class uses IterativelySubtractedPSFPhotometry, but is more specialized in that it implements the DAOPHOT-specific star-finder, grouper, and subtraction routines". or something like that